### PR TITLE
Starts work on event trigger feature

### PR DESCRIPTION
**Pull request by stevsmit**
https://issues.redhat.com/browse/PROJQUAY-6977

### CODE REVIEW

File: modules/rn_3_12_0.adoc [hunk 1]
```diff
@@ -30,9 +30,14 @@ The following documentation changes have been made with the {productname} {produ
 
 The following updates have been made to {productname}.
 
-[id=""]
+[id="image-expiration-event"]
+=== Image expiration notification
 
-=== <Example_Feature>
+Previously, {productname} administrators and users had no way of being alerted when an image was about to expire. With this update, an event can be configured to notify users when an image is about to expire, or has expired. This helps {productname} users avoid unexpected pull failures.
```
Comment (Sunandadadi) [inline modules/rn_3_12_0.adoc:36]: We only track images that are going to expire. We don't notify about images that are already expired.

File: modules/creating-image-expiration-notification.adoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// * use_quay/master.adoc
+// * quay_io/master.adoc
+
+:_content-type: CONCEPT
+[id="creating-image-expiration-notification"]
+= Creating an image expiration notification
+
+Image expiration event triggers can be configured to notify users through email, Slack webhooks, and so on, and can be configured at the repository level. Triggers can be set for images expiring in any amount of days, and can work in conjunction with the auto-pruning feature.
+
+Image expiration notifications can only be set by using the {productname} v2 UI.
```
Comment (Sunandadadi) [inline modules/creating-image-expiration-notification.adoc:12]: We do have an existing `createRepoNotification` endpoint that can be used to create too.

**Review by Sunandadadi — CHANGES_REQUESTED** [review]

File: modules/creating-image-expiration-notification.adoc [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+. Click *Submit*. You are returned to the *Events and notifications* page, and the notification now appears.
+
+. Optional. You can set the `NOTIFICATION_TASK_RUN_MINIMUM_INTERVAL_MINUTES` variable in your `config.yaml` file if you want to notifications for expiring images to re-run automatically. For example:
++
+[source,yaml]
+----
+NOTIFICATION_TASK_RUN_MINIMUM_INTERVAL_MINUTES: 60 <1>
+----
+<1> By setting this field to `60`, you will receive a notification about upcoming events every hour.
```
Comment (Sunandadadi) [inline modules/creating-image-expiration-notification.adoc:54]: only if there are an any updates.

File: modules/creating-image-expiration-notification.adoc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+. Depending on which method you chose, include the necessary data. For example, if you chose *Webhook POST*, include the `Webhook URL`. 
+
+. Optional. Provide a *POST JSON body template*.
+
+. Optional. Provide a *Title* for your notification.
+
+. Click *Submit*. You are returned to the *Events and notifications* page, and the notification now appears.
+
+. Optional. You can set the `NOTIFICATION_TASK_RUN_MINIMUM_INTERVAL_MINUTES` variable in your `config.yaml` file if you want to notifications for expiring images to re-run automatically. For example:
```
Comment (Sunandadadi) [inline modules/creating-image-expiration-notification.adoc:48]: how about something like - if you want to over write the existing re-run default which is 5hrs?

File: modules/config-fields-tag-expiration.adoc [hunk 5]
```diff
@@ -29,6 +29,10 @@ The following tag expiration configuration fields are available with {productnam
 | **FEATURE_AUTO_PRUNE** | Boolean | When set to `True`, enables functionality related to the auto-pruning of tags. 
  +
 *Default:* `False`
+
+| *NOTIFICATION_TASK_RUN_MINIMUM_INTERVAL_MINUTES* |Integer | The interval, in minutes, that defines the frequency to re-run notifications for expiring images. +
```
Comment (Sunandadadi) [inline modules/config-fields-tag-expiration.adoc:33]: can you also add that this defaults to 5hrs?

File: modules/creating-image-expiration-notification.adoc [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+[id="creating-image-expiration-notification"]
+= Creating an image expiration notification
+
+Image expiration event triggers can be configured to notify users through email, Slack webhooks, and so on, and can be configured at the repository level. Triggers can be set for images expiring in any amount of days, and can work in conjunction with the auto-pruning feature.
+
+Image expiration notifications can be set by using the {productname} v2 UI or by using the `createRepoNotification` API endpoint.
+
+.Prerequisites
+
+* You have set `FEATURE_GARBAGE_COLLECTION: true` in your `config.yaml` file.
```
Comment (SeanZhao-redhat) [inline modules/creating-image-expiration-notification.adoc:16]: optional. as by default it is true https://github.com/quay/quay/blob/a06b6e929c7a38b78ba885fa2dfc425f6b995dc1/config.py#L729
@Sunandadadi wdyt?

Comment (SeanZhao-redhat) [reply modules/creating-image-expiration-notification.adoc:16]: instead,  `FEATURE_AUTO_PRUNE: true` looks a must

File: modules/creating-image-expiration-notification.adoc [hunk 7]
```diff
@@ -0,0 +1,82 @@
+// module included in the following assemblies:
+
+// * use_quay/master.adoc
+// * quay_io/master.adoc
+
+:_content-type: CONCEPT
+[id="creating-image-expiration-notification"]
+= Creating an image expiration notification
+
+Image expiration event triggers can be configured to notify users through email, Slack webhooks, and so on, and can be configured at the repository level. Triggers can be set for images expiring in any amount of days, and can work in conjunction with the auto-pruning feature.
```
Comment (SeanZhao-redhat) [inline modules/creating-image-expiration-notification.adoc:10]: a comma expected:  Slack, Webhooks

File: modules/creating-image-expiration-notification.adoc [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+. In the *When the image is due to expiry in days* box, enter the number of days before the image's expiration when you want to receive an alert. For example, use `1` for 1 day.
+
+. In the *Select method...* box, click one of the following:
++
+* E-mail
+* Webhook POST
+* Flowdock Team Notification
+* HipChat Room Notification
+* Slack Room Notification
```
Comment (SeanZhao-redhat) [inline modules/creating-image-expiration-notification.adoc:38]: should be: Slack Notification, notification to a slack Channel not Room

File: modules/creating-notifications.adoc [hunk 9]
```diff
@@ -25,26 +25,27 @@ endif::[]
...
 
 . After you have selected the event type, select the notification method. The following  methods are supported:
 +
 * Quay Notification
-* E-mail
+* E-mail Notification
 * Webhook POST
 * Flowdock Team Notification
 * HipChat Room Notification
 * Slack Room Notification
```
Comment (SeanZhao-redhat) [inline modules/creating-notifications.adoc:47]: same, should be: Slack Notification

File: modules/rn_3_12_0.adoc [hunk 10]
```diff
@@ -39,6 +39,15 @@ With this release, organization administrators can now assign OAuth API tokens t
 
 For more information, see link:https://docs.redhat.com/en/documentation/red_hat_quay/{producty}/html-single/use_red_hat_quay/index#reassigning-oauth-access-token[Reassigning an OAuth access token].
 
+[id="image-expiration-event"]
+=== Image expiration notification
+
+Previously, {productname} administrators and users had no way of being alerted when an image was about to expire. With this update, an event can be configured to notify users when an image is about to expire. This helps {productname} users avoid unexpected pull failures.
+
+Image expiration event triggers can be configured to notify users through email, Slack webhooks, and so on, and can be configured at the repository level. Triggers can be set for images expiring in any amount of days, and can work in conjunction with the auto-pruning feature.
```
Comment (SeanZhao-redhat) [inline modules/rn_3_12_0.adoc:47]: same, a comma expected: Slack, Webhooks

File: modules/creating-image-expiration-notification.adoc (hunk 6 above)
Comment (stevsmit) [reply modules/creating-image-expiration-notification.adoc:16]: Added/slightly reworded.

Comment (Sunandadadi) [reply modules/creating-image-expiration-notification.adoc:16]: @SeanZhao-redhat 
`FEATURE_GARBAGE_COLLECTION` is true by default but it is a must for this feature. 
`FEATURE_AUTO_PRUNE` is not a must.

**Pull request merged**